libnavigation-core/src/test/java/com/mapbox/navigation/core/replay/route2/ReplayRouteSmootherTest.kt

```diff
@@ -0,0 +1,158 @@
+package com.mapbox.navigation.core.replay.route2
+
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertTrue
+import org.junit.Assert
+import org.junit.Test
+
+class ReplayRouteSmootherTest {
+
+    private val recommendedThresholdMeters = 3.0
+
+    private val routeSmoother = ReplayRouteSmoother()
+
+    @Test
+    fun `should turn straight road into a segment`() {
+        // Geometry used qnq_gAxdhmhFuvBlJe@?qC^
+        val coordinates = listOf<Point>(
+            Point.fromLngLat(-122.393181, 37.758201),
+            Point.fromLngLat(-122.393364, 37.760116),
+            Point.fromLngLat(-122.393364, 37.760135),
+            Point.fromLngLat(-122.39338, 37.760208)
+        )
+
+        val smoothedRoutes = routeSmoother.smoothRoute(coordinates, recommendedThresholdMeters)
+
+        assertEquals(2, smoothedRoutes.size)
+        assertEquals(-122.393181, smoothedRoutes[0].point.longitude())
+        assertEquals(37.758201, smoothedRoutes[0].point.latitude())
+        assertEquals(-122.39338, smoothedRoutes[1].point.longitude())
+        assertEquals(37.760208, smoothedRoutes[1].point.latitude())
+    }
+
+    @Test
+    fun `should turn a turn into three points`() {
+        // Geometry used anq_gAxdhmhFbZkA^?tDMUsF?m@WmKMoHOeF]eO?}@GiBcB}s@?{@McGoDLu@?cUlAqMj@qfAtE
```

Comment:
Same as above re: `coordinates` and geometry used

Reply:
yeah, removed since the coordinates are all there